packages/product/src/loaders/connection.ts

```diff
@@ -6,52 +6,73 @@ import {
   MODULE_RESOURCE_TYPE,
   MODULE_SCOPE,
 } from "@medusajs/modules-sdk"
-import { DALUtils, MedusaError, ModulesSdkUtils } from "@medusajs/utils"
+import {
+  DALUtils,
+  MedusaError,
+  ModulesSdkUtils,
+  ContainerRegistrationKeys,
+} from "@medusajs/utils"
 
 import { EntitySchema } from "@mikro-orm/core"
 
 import * as ProductModels from "@models"
-import { ConfigModule, ModulesSdkTypes } from "@medusajs/types"
+import { PostgreSqlDriver, SqlEntityManager } from "@mikro-orm/postgresql"
+import { ModulesSdkTypes } from "@medusajs/types"
 
 export default async (
   {
     options,
     container,
+    logger,
   }: LoaderOptions<
     | ModulesSdkTypes.ModuleServiceInitializeOptions
     | ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
   >,
   moduleDeclaration?: InternalModuleDeclaration
 ): Promise<void> => {
+  let manager = (
+    options as ModulesSdkTypes.ModuleServiceInitializeCustomDataLayerOptions
+  )?.manager
+
   if (
     moduleDeclaration?.scope === MODULE_SCOPE.INTERNAL &&
-    moduleDeclaration.resources === MODULE_RESOURCE_TYPE.SHARED
+    moduleDeclaration.resources === MODULE_RESOURCE_TYPE.SHARED &&
+    !manager
```

Comment:
suggestion: would be good to add a comment here on why it `!manager` is important

Reply:
Thought it was explicit, if the user passes a manager we don't want to create a new one nor resolve something we won't use. I ll add a comment

Reply:
If thats the case, shouldn't it throw an error? 

If its a shared resource, the module shouldn't have to be the one creating it right?

Reply:
It is just that since it is something the user can pass through the options, it just apply a precedence order, in fact, at the beginning we could just return early if there is a manager passed as an option

Reply:
Yes, exactly. The manager being present or not doesn't seem to relate to the shared or internal commands. Returning early with just the manager scope makes sense.